Makefile

```diff
@@ -0,0 +1,69 @@
+-s -w"
+
+all: clean build-all
+
+build-all: clean auth-provider-gcp-linux-amd64 auth-provider-gcp-linux-arm64 auth-provider-gcp-windows-amd64 \
+	cloud-controller-manager-linux-amd64 cloud-controller-manager-linux-arm64 \
+	gke-gcloud-auth-plugin-linux-amd64 gke-gcloud-auth-plugin-linux-arm64 gke-gcloud-auth-plugin-windows-amd64 gke-gcloud-auth-plugin-windows-arm64	gke-gcloud-auth-plugin-darwin-arm64
+
+cloud-controller-manager-linux-amd64 cloud-controller-manager-linux-arm64: cloud-controller-manager-linux-%:
+	mkdir -p release/$(GIT_VERSION)/cloud-controller-manager/linux/$*
+	CGO_ENABLED=0 GOOS=linux GOARCH=$* go build $(LDFLAGS) -o release/$(GIT_VERSION)/cloud-controller-manager/linux/$*/cloud-controller-manager k8s.io/cloud-provider-gcp/cmd/cloud-controller-manager
+
+auth-provider-gcp-linux-amd64 auth-provider-gcp-linux-arm64: auth-provider-gcp-linux-%:
+	mkdir -p release/$(GIT_VERSION)/auth-provider-gcp/linux/$*
+	CGO_ENABLED=0 GOOS=linux GOARCH=$* go build $(LDFLAGS) -o release/$(GIT_VERSION)/auth-provider-gcp/linux/$*/auth-provider-gcp k8s.io/cloud-provider-gcp/cmd/auth-provider-gcp
+
+auth-provider-gcp-windows-amd64:
+	mkdir -p release/$(GIT_VERSION)/auth-provider-gcp/windows/amd64
+	CGO_ENABLED=0 GOOS=windows GOARCH=amd64 go build $(LDFLAGS) -o release/$(GIT_VERSION)/auth-provider-gcp/windows/amd64/auth-provider-gcp.exe k8s.io/cloud-provider-gcp/cmd/auth-provider-gcp
+
+gke-gcloud-auth-plugin-linux-amd64 gke-gcloud-auth-plugin-linux-arm64: gke-gcloud-auth-plugin-linux-%:
+	mkdir -p release/$(GIT_VERSION)/gke-gcloud-auth-plugin/linux/$*
+	CGO_ENABLED=0 GOOS=linux GOARCH=$* go build $(LDFLAGS) -o release/$(GIT_VERSION)/gke-gcloud-auth-plugin/linux/$*/gke-gcloud-auth-plugin k8s.io/cloud-provider-gcp/cmd/gke-gcloud-auth-plugin
+
+gke-gcloud-auth-plugin-windows-amd64 gke-gcloud-auth-plugin-windows-arm64: gke-gcloud-auth-plugin-windows-%:
+	mkdir -p release/$(GIT_VERSION)/gke-gcloud-auth-plugin/windows/$*
+	CGO_ENABLED=0 GOOS=windows GOARCH=$* go build $(LDFLAGS) -o release/$(GIT_VERSION)/gke-gcloud-auth-plugin/windows/$*/gke-gcloud-auth-plugin.exe k8s.io/cloud-provider-gcp/cmd/gke-gcloud-auth-plugin
+
+gke-gcloud-auth-plugin-darwin-arm64:
+	mkdir -p release/$(GIT_VERSION)/gke-gcloud-auth-plugin/darwin/arm64
+	CGO_ENABLED=0 GOOS=darwin GOARCH=arm64 go build $(LDFLAGS) -o release/$(GIT_VERSION)/gke-gcloud-auth-plugin/darwin/arm64/gke-gcloud-auth-plugin k8s.io/cloud-provider-gcp/cmd/gke-gcloud-auth-plugin
+
+
+.PHONY: all clean build-all copy-binaries-to-gcs
+
+clean:
+	@echo "Cleaning up..."
+	@find release/ -type d -mindepth 1 -print0 | xargs -0 rm -rf
+
+release-tars:
+	bazel build release:release-tars
+
+copy-binaries-to-gcs: build-all
+	gcloud storage cp --recursive release/$(GIT_VERSION)/auth-provider-gcp gs://$(BUCKET_NAME)/auth-provider-gcp/$(GIT_VERSION)
+	gcloud storage cp --recursive release/$(GIT_VERSION)/gke-gcloud-auth-plugin gs://$(BUCKET_NAME)/gke-gcloud-auth-plugin/$(GIT_VERSION)
```

Comment:
before this was

 ```
      - cp
      - /workspace/bazel-base-windows-amd64/execroot/io_k8s_cloud_provider_gcp/bazel-out/k8-fastbuild/bin/cmd/gke-gcloud-auth-plugin/gke-gcloud-auth-plugin_/gke-gcloud-auth-plugin.exe
      - gs://k8s-staging-cloud-provider-gcp/gke-gcloud-auth-plugin/windows-amd64/${_GIT_TAG}
 ```

 , this is not the same because the layout changed `release/v34.0.0-18-g73276375b/gke-gcloud-auth-plugin/windows/amd64/gke-gcloud-auth-plugin.exe` 

I also think we should be more explicit in the copy and not copy recursively

Reply:
Recursive copying is fine as we clean, build and then copy to the bucket. Also, it is an ephemeral build env, so the directory is already empty and clean.

Reply:
and the uploaded one?